src/encoded/static/components/static-pages/placeholders/ExperimentSetMatrix.js

```diff
@@ -0,0 +1,600 @@
+            //"data_source"               : "Available through",
+            //"award"                     : "Award",
+            //"accession"                 : "Accession",
+            //"number_of_experiments"     : "# Experiments in Set",
+            //"submitted_by"              : "Submitter",
+            //"experimentset_type"        : "Set Type",
+        },
+        "columnSubGroupingOrder"    : ["Submitted", "In Submission", "Planned", "Not Planned"]
+    };
+
+    static propTypes = {
+        'sectionKeys': PropTypes.arrayOf(PropTypes.string).isRequired,
+        'queries': PropTypes.object.isRequired,
+        'valueChangeMap': PropTypes.object,
+        'fieldChangeMap': PropTypes.object,
+        'groupingProperties': PropTypes.object,
+        'columnGrouping': PropTypes.object,
+        'headerFor': PropTypes.object,
+        'sectionStyle': PropTypes.object,
+        'fallbackNameForBlankField': PropTypes.string,
+        'statePrioritizationForGroups': PropTypes.arrayOf(PropTypes.string),
+        'headerPadding': PropTypes.number,
+        'headerColumnsOrder': PropTypes.arrayOf(PropTypes.string),
+        'titleMap': PropTypes.object,
+        'columnSubGroupingOrder': PropTypes.arrayOf(PropTypes.string)
+    }
+
+    static convertResult(result, dataSource, fieldChangeMap, valueChangeMap, statusStateTitleMap, fallbackNameForBlankField){
+
+        const convertedResult = _.clone(result);
+
+        if (fieldChangeMap[dataSource]){
+            _.forEach(_.pairs(fieldChangeMap[dataSource]), function([ fieldToMapTo, fieldToMapFrom ]){
+                let value = object.getNestedProperty(result, fieldToMapFrom, fieldToMapTo);
+                if (Array.isArray(value)){ // Only allow single vals.
+                    value = _.uniq(_.flatten(value));
+                    if (value.length > 1){
+                        console.warn("We have 2+ of a grouping value", fieldToMapFrom, value, result);
+                    }
+                    value = value[0] || fallbackNameForBlankField;
+                }
+                convertedResult[fieldToMapTo] = value;
+            }, {});
+        }
+
+        // Change values (e.g. shorten some):
+        if (valueChangeMap[dataSource]){
+            _.forEach(_.pairs(valueChangeMap[dataSource]), function([field, changeMap]){
+                if (typeof convertedResult[field] === "string"){ // If present
+                    convertedResult[field] = changeMap[convertedResult[field]] || convertedResult[field];
+                }
+            });
+        }
+
+        // Standardized state from status
+        // TODO Use similar by-data-source structure as fieldChangeMap & valueChangeMap
+        if (statusStateTitleMap){
+            const [ stateTitleToSave ] = _.find(_.pairs(statusStateTitleMap), function([titleToSave, validStatuses]){ return validStatuses.indexOf(result.status) > -1; });
+            convertedResult.state = stateTitleToSave || fallbackNameForBlankField;
```

Comment:
Codacy found an issue: ['statusStateTitleMap' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5723113)